template-only-docs/template-technical-design-principles.md

```diff
@@ -0,0 +1,48 @@
+# Template technical design principles
+
+This document outlines the technical design principles that the template-infra project follows. These principles guide the development of the infrastructure template.
+
+## Design for simplicity
+
+### Design for the common case
+
+- Optimize for the common case
+- Avoid adding unnecessary flexibility that increases complexity
+- Ensure that uncommon cases can be addressed through explicit customization
```

Comment:
Say more? Is the idea that we should design for the common use case BUT also explicitly call-out when we are customizing things for uncommon cases? Is this implying documentation, code architecture, or something else? 

I guess I'm mostly trying to figure out how this line aligns with the other two

Reply:
I think the idea is to still try to adopt good design practices that makes things easy to extend and refactor so that project teams can modify things relatively easily.

> explicitly call-out when we are customizing things for uncommon cases

No I don't mean this. The stuff in the template isn't customized by definition, since it's part of the template. I'm defining customization to mean modifying what's in the template.